.azure/linux-cabal.yml

```diff
@@ -0,0 +1,75 @@
+jobs:
+- job: Linux_Cabal
+  timeoutInMinutes: 0
+  pool:
+    vmImage: ubuntu-16.04
+  strategy:
+    matrix:
+      ghc-8.6.5:
+        GHC_VERSION: "8.6.5"
+      ghc-8.4.4:
+        GHC_VERSION: "8.4.4"
+  variables:
+    CABAL_VERSION: "3.0"
+  steps:
+  - task: Cache@2
+    inputs:
+      key: '"cabal-store" | "$(Agent.OS)" | "$(CABAL_VERSION)" | "$(GHC_VERSION)" | $(Build.SourcesDirectory)/cabal.project | $(Build.SourcesDirectory)/haskell-ide-engine.cabal | $(Build.SourcesDirectory)/hie-plugin-api/hie-plugin-api.cabal'
+      path: .azure-cache
+      cacheHitVar: CACHE_RESTORED
+    displayName: "Download cache"
+  - bash: |
+      mkdir -p $HOME/.cabal
+      tar -vxzf .azure-cache/cabal-root.tar.gz -C /
+      mkdir -p $HOME/.ghc
+      tar -vxzf .azure-cache/ghc-root.tar.gz -C /
+      mkdir -p dist-newtyle
+      tar -vxzf .azure-cache/cabal-dist.tar.gz
+    displayName: "Unpack cache"
+    condition: eq(variables.CACHE_RESTORED, 'true')
+  - bash: |
+      git submodule sync
+      git submodule update --init
+    displayName: Sync submodules
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-update
+      cabal v2-build --only-dependencies
+    displayName: Build dependencies
+  - bash: |
+      source .azure/linux-cabal.bashrc
+      cabal v2-build
+    displayName: Build `hie`
```

Comment:
As we have seen in #1562, this does not seem to be necessary. However, it is a minor reduction in time-consumption, thus, nothing to worry about or even necessary to change, imo.

Reply:
Yeah, only to make it explicit as a separate step that will fail if the code has some compile time error